test/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDaoTest.java

```diff
@@ -130,11 +132,43 @@ public void createUpdateDeleteActivities() throws Exception {
         
         activities = activityDao.getActivities(context);
         assertEquals("deleted activity not returned from server", collectionSize-1, activities.size());
-        activityDao.deleteActivities(user.getHealthCode());
+        activityDao.deleteActivitiesForUser(user.getHealthCode());
         
         activities = activityDao.getActivities(context);
         assertEquals("all activities deleted", 0, activities.size());
     }
+    
+    @Test
+    public void createActivitiesAndDeleteSomeBySchedulePlan() throws Exception {
+        DateTime endsOn = DateTime.now().plus(Period.parse("P4D"));
+        Map<String,DateTime> events = Maps.newHashMap();
+        events.put("enrollment", ENROLLMENT);
+        
+        ScheduleContext context = new ScheduleContext.Builder()
+            .withStudyIdentifier(TEST_STUDY)
+            .withClientInfo(ClientInfo.UNKNOWN_CLIENT)
+            .withTimeZone(DateTimeZone.UTC)
+            .withEndsOn(endsOn)
+            .withHealthCode(user.getHealthCode())
+            .withEvents(events).build();
+        
+        List<ScheduledActivity> activities = TestUtils.runSchedulerForActivities(user, context);
+        activityDao.saveActivities(activities);
+
+        // Get one schedule plan GUID to delete and the initial count
+        String schedulePlanGuid = activities.get(0).getSchedulePlanGuid();
```

Comment:
This is dangerous, as this schedulePlanGuid might come from an unrelated test. It would be safer to create a known schedule plan up front so we know what the schedule plan guid is, and we can target that specific schedule.

One side benefit is, if we control the schedule plan, we can use absolute counts instead of regular counts for this test.